assets/php/virion.php

```diff
@@ -61,6 +61,15 @@ function virion_infect(Phar $virus, Phar $host, string $prefix = "", int $mode =
     if(!is_array($virionYml)) {
         throw new RuntimeException("Corrupted virion.yml, could not activate virion", 2);
     }
+    $srcNamespacePrefix = "";
+    if(isset($host["plugin.yml"])) {
+        $hostYml = yaml_parse(file_get_contents($host["plugin.yml"]));
+        if(!is_array($hostYml)) {
+            throw new RuntimeException("Corrupted plugin.yml in host plugin, could not inject virion", 2);
+        }
+        $host["plugin.yml"] = yaml_emit($hostYml);
```

Comment:
why do we need to write back?

Reply:
Otherwise it will make the file blank (after reading I assume)

Reply:
Goes blank with no write back, see attached
[host.phar.zip](https://github.com/poggit/poggit/files/9280913/host.phar.zip)

Reply:
I could not reproduce it in my environment.
Could it be a bug in PHP?

Reply:
More likely some disk error (e.g. out of space).

Reply:
> More likely some disk error (e.g. out of space).

Definitely not unfortunately, will get a simpler reproduction file rather than the entire poggit setup

Reply:
@dktapps or @SOF3 you guys will probably know a lot more than me in this area, but _WITHOUT_ the write back *AND* `$host->stopBuffering()` it will clear the file, but _WITHOUT_ the write back and without the stopBuffering it wont clear.

Reply:
See https://pastebin.com/Qx4NwnYL for a reproducible script excluding poggit junk